x-pack/plugins/security_solution/public/siem_migrations/rules/hooks/use_filter_rules_to_install.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import type { RuleMigration } from '../../../../common/siem_migrations/model/rule_migration.gen';
+import type { FilterOptions } from '../../../detection_engine/rule_management/logic/types';
+
+export type TableFilterOptions = Pick<FilterOptions, 'filter'>;
+
+export const useFilterRulesToInstall = ({
```

Comment:
We'll need to do this on the ES side once we add pagination, but it's ok for the initial PR